src/components/Layout/Header/hooks.ts

```diff
@@ -0,0 +1,75 @@
+import { useMemo, useState } from 'react'
+import { useLocation } from 'react-router'
+
+import { HeaderProps } from 'components/Layout/Header'
+import { HeaderNavLinksInterface } from 'components/Layout/Header/Navigation'
+
+import useWindowSizes from 'hooks/useWindowSizes'
+
+interface UseNavigationProps {
+  navigationArray: HeaderNavLinksInterface[]
+  handleLinkSelect: (key: string) => void
+}
+
+export const useNavigation = (initialState: HeaderProps['navigation'], responsive: boolean): UseNavigationProps => {
+  const [navObject, setNavObject] = useState(initialState)
+  const location = useLocation()
+
+  useMemo(() => {
+    if (responsive) {
+      const smartNavObject = Object.keys(initialState).reduce((acc, key, index) => {
+        const value = initialState[key]
+        const editedHeaderValue = {
+          ...value,
+          order: location.pathname.search(key.toLowerCase()) === 1 ? 0 : index + 1,
+        }
+        const newAcc = {
+          ...acc,
+          [key]: { ...editedHeaderValue },
+        }
+
+        return newAcc
+      }, {})
+
+      setNavObject(smartNavObject)
+    }
+  }, [initialState, location.pathname, responsive])
+
+  const navigationArray: HeaderNavLinksInterface[] = useMemo(
+    () =>
+      Object.keys(navObject).reduce((accum: HeaderNavLinksInterface[], currentKey) => {
+        const navArray = [...accum, navObject[currentKey]]
+        return navArray
+      }, []),
+    [navObject],
+  )
+
+  const handleLinkSelect = (key: string): void =>
```

Comment:
`useCallback`?

Reply:
so in this case the savings are actually worse - seeing as using callback can be rewritten as:
```
const linkSelect = (key: string): void =>
    setNavObject(prevState => ({
      ...initialState,
      [key]: {
        ...prevState[key],
        order: 0,
      },
    }))

const handleLinkSelect = useCallback(linkSelect, [])
```
actually probably has to do more work here as it's calling `useCallback` and instantiating the array. not 100% but i think it's better as is since it's not such an expensive function

same thing for `useMemo`